via/services/http.py

```diff
@@ -83,34 +83,40 @@ def request(self, method, url, timeout=(10, 10), **kwargs):
                 **kwargs,
             )
             response.raise_for_status()
+
         except Exception as err:
-            if self._error_translator and (mapped := self._error_translator(err)):
-                raise mapped from err
-
-            if not isinstance(err, RequestException):
-                # If the error_translator didn't handle a non request exception, raise it directly
-                raise err
-
-            translated_class = UnhandledUpstreamException
-            for (  # pragma: no branch
-                error_class,
-                target_class,
-            ) in DEFAULT_ERROR_MAP.items():
-                if isinstance(err, error_class):
-                    translated_class = target_class
-                    break
-
-            raise translated_class(
-                message=err.args[0] if err.args else None,
-                requests_err=err if hasattr(err, "request") else None,
-            ) from err
+            raise self._translate_exception(err) from err
 
         return response
 
     def stream(self, url, method="GET", **kwargs):
         response = self.request(method=method, url=url, stream=True, **kwargs)
-
-        yield from self._stream_bytes(response)
+        try:
+            yield from self._stream_bytes(response)
+        except Exception as err:
+            raise self._translate_exception(err) from err
```

Comment:
There's a slight change of behavior here from the original method. It's possible for `err` to not end up being mapped. In which case we end up doing `raise err from err` which will duplicate the error in all the reporting (and possibly be a bit confusing).

In the original formulation it was something like:


```suggestion
            if mapped_err := self._translate_exception(err):
                raise self._translate_exception(err) from err

            raise
```

You'd want to do this in the one above as well.